ml-agents/mlagents/trainers/trainer.py

```diff
@@ -152,19 +151,19 @@ def increment_step(self, n_steps: int) -> None:
 
         :param n_steps: number of steps to increment the step count by
         """
-        self.step = self.policy.increment_step(n_steps)
+        self.step += n_steps
```

Comment:
Are we still incrementing the policy's steps somewhere? I believe the steps are needed to update the nodes in the TF graph that hold the global steps, which are used for annealing the different params

Reply:
Yes, the policy steps get updated right after this is called in trainer_controller.

Reply:
Is there a reason why we don't just have `increment_step(self, n_steps, behavior_name_id)` and do the get_policy inside the method? So we don't have to keep track of two step increments

Reply:
Basically in my latest code (https://github.com/Unity-Technologies/ml-agents/commit/28ad94a733e2dc79cea3a59521804e2e52350a6e) I've replaced all of these calls on the trainer with a single `advance()` call that takes care of the incrementing, trajectory ingestion, and policy updates.